refactor(npm): Migrate package to @decentdao/decent-contracts with new publishing system

adamgall opened the pull request:
Depends on https://github.com/decentdao/decent-contracts/pull/402

Fixes [ENG-1229](https://linear.app/decent-labs/issue/ENG-1229/migrate-npm-package-to-decentdaodecent-contracts-with-new-publishing)

## Motivation

Complete the migration from @fractal-framework/fractal-contracts to @decentdao/decent-contracts with a new publishing system that supports both legacy hardhat-deploy and new hardhat-ignition deployment artifacts. This ensures backward compatibility while enabling the new CREATE2 deployment strategy.

## Change Summary

- Renamed npm package from @fractal-framework/fractal-contracts to @decentdao/decent-contracts
- Bumped version to 2.0.0 to reflect the major migration
- Updated security contact from security@decent-dao.org to security@decentlabs.io
- Updated GitHub issue links to new repository structure
- Rewrote prepublish.sh script to use new TypeScript processing pipeline
- Created process-deployments.ts script that:
  - Processes both legacy and ignition deployment artifacts
  - Groups contracts by category dynamically
  - Validates cross-network deployment consistency
  - Outputs typed exports for both current and legacy deployments
- Updated README deployment section with hardhat-ignition commands
- Added documentation for CREATE2 salt configuration

The new system maintains full backward compatibility through the legacy exports while providing a cleaner, category-based structure for new deployments.

tbwebb22 reviewed (changes requested):
Search codebase for 'Fractal' and remove it

tbwebb22 commented on scripts/process-deployments.ts:
```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/env ts-node
```
could add `supported networks`

adamgall replied:
done, pending push

adamgall committed:
refactor(npm): Migrate package to @decentdao/decent-contracts with new publishing system

/## Motivation

Complete the migration from @fractal-framework/fractal-contracts to @decentdao/decent-contracts
with a new publishing system that supports both legacy hardhat-deploy and new hardhat-ignition
deployment artifacts. This ensures backward compatibility while enabling the new CREATE2
deployment strategy.

/## Change Summary

- Renamed npm package from @fractal-framework/fractal-contracts to @decentdao/decent-contracts
- Bumped version to 2.0.0 to reflect the major migration
- Updated security contact from security@decent-dao.org to security@decentlabs.io
- Updated GitHub issue links to new repository structure
- Rewrote prepublish.sh script to use new TypeScript processing pipeline
- Created process-deployments.ts script that:
  - Processes both legacy and ignition deployment artifacts
  - Groups contracts by category dynamically
  - Validates cross-network deployment consistency
  - Outputs typed exports for both current and legacy deployments
- Updated README deployment section with hardhat-ignition commands
- Added documentation for CREATE2 salt configuration

The new system maintains full backward compatibility through the legacy exports while
providing a cleaner, category-based structure for new deployments.

adamgall commented:
@tbwebb22 this is ready for another review. i've removed all instances of "fractal" from the codebase (except for `ModuleFractal` of course), and added a `chains` property to the published index file

https://www.npmjs.com/package/@decentdao/decent-contracts?activeTab=code

<img width="1229" alt="Screenshot 2025-06-23 at 7 24 52 PM" src="https://github.com/user-attachments/assets/5981b35b-1e16-4a4d-850a-c3bbc30d0cea" />
